src/test/java/edu/harvard/iq/dataverse/api/FilesIT.java

```diff
@@ -1403,68 +1400,271 @@ public void testDataSizeInDataverse() throws InterruptedException {
         
     }
-    
+
     @Test
     public void testGetFileInfo() {
-
         Response createUser = UtilIT.createRandomUser();
-        String username = UtilIT.getUsernameFromResponse(createUser);
-        String apiToken = UtilIT.getApiTokenFromResponse(createUser);
-        Response makeSuperUser = UtilIT.makeSuperUser(username);
-        String dataverseAlias = createDataverseGetAlias(apiToken);
-        Integer datasetId = createDatasetGetId(dataverseAlias, apiToken);
+        String superUserUsername = UtilIT.getUsernameFromResponse(createUser);
+        String superUserApiToken = UtilIT.getApiTokenFromResponse(createUser);
+        UtilIT.makeSuperUser(superUserUsername);
+        String dataverseAlias = createDataverseGetAlias(superUserApiToken);
+        Integer datasetId = createDatasetGetId(dataverseAlias, superUserApiToken);
 
         createUser = UtilIT.createRandomUser();
-        String apiTokenRegular = UtilIT.getApiTokenFromResponse(createUser);
+        String regularUsername = UtilIT.getUsernameFromResponse(createUser);
+        String regularApiToken = UtilIT.getApiTokenFromResponse(createUser);
 
         msg("Add a non-tabular file");
         String pathToFile = "scripts/search/data/binary/trees.png";
-        Response addResponse = UtilIT.uploadFileViaNative(datasetId.toString(), pathToFile, apiToken);
+        Response addResponse = UtilIT.uploadFileViaNative(datasetId.toString(), pathToFile, superUserApiToken);
 
+        // The following tests cover cases where no version ID is specified in the endpoint
+        // Superuser should get to see draft file data
         String dataFileId = addResponse.getBody().jsonPath().getString("data.files[0].dataFile.id");
-        msgt("datafile id: " + dataFileId);
-
-        addResponse.prettyPrint();
-
-        Response getFileDataResponse = UtilIT.getFileData(dataFileId, apiToken);
-
-        getFileDataResponse.prettyPrint();
+        Response getFileDataResponse = UtilIT.getFileData(dataFileId, superUserApiToken);
+        String newFileName = "trees.png";
         getFileDataResponse.then().assertThat()
-                .body("data.label", equalTo("trees.png"))
-                .body("data.dataFile.filename", equalTo("trees.png"))
+                .body("data.label", equalTo(newFileName))
+                .body("data.dataFile.filename", equalTo(newFileName))
                 .body("data.dataFile.contentType", equalTo("image/png"))
                 .body("data.dataFile.filesize", equalTo(8361))
                 .statusCode(OK.getStatusCode());
-        
-        getFileDataResponse = UtilIT.getFileData(dataFileId, apiTokenRegular);
+
+        // Regular user should not get to see draft file data
+        getFileDataResponse = UtilIT.getFileData(dataFileId, regularApiToken);
         getFileDataResponse.then().assertThat()
-                .statusCode(BAD_REQUEST.getStatusCode());
+                .statusCode(UNAUTHORIZED.getStatusCode());
```

Comment:
If you're sending a valid api token, I think this should be FORBIDDEN (403) rather than UNAUTHORIZED (401).

Reply:
I agree, but this is a core behavior in the API. When the user does not have sufficient permissions to execute the commands associated with an endpoint, a 401 is always returned. 

See `execCommand` method from `AbstractApiBean` (https://github.com/IQSS/dataverse/blob/develop/src/main/java/ edu/harvard/iq/dataverse/api/AbstractApiBean.java#L604).

Changing the error code should be a general change IMO to be consistent across the API.

Reply:
I created #10340 for a more general fix.